[Android] Build issue with node: Couldn't follow symbolic link

Under circumstances that are not clear, the WPAndroid build fails with this error:

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':react-native-gutenberg-bridge:nodeSetup'.
> Could not list contents of '/var/folders/d3/7bzlyz453w11k42t26xtbkmc0000gn/T/jsbundle/null/nodejs/node-v12.16.1-darwin-x64/bin/npm'. Couldn't follow symbolic link.
```

A simliar issue can happen with Yarn:

```
Could not list contents of ‘/var/folders/py/vdkc920j69b255sld66fbxt40000gp/T/jsbundle/null/yarn/yarn-v1.22.4/bin/yarnpkg’. Couldn’t follow symbolic link.
```

Looking at the symlinks in question, they point to files that are no longer there. We don't know yet why those files get missing and the tools setup becomes inconsistent.

--- a/react-native-gutenberg-bridge/android/build.gradle
+++ b/react-native-gutenberg-bridge/android/build.gradle
@@ -33,6 +33,11 @@ def buildGutenbergMobileJSBundle =
         ? false : (rootProject.ext.has("buildGutenbergMobileJSBundle")
                 && rootProject.ext.buildGutenbergMobileJSBundle)
 
+def tempFolder = "${System.getenv('TMPDIR')}/jsbundle/${System.getenv('VERSION')}"
+def nodeFolder = file("${tempFolder}/nodejs")
+def npmFolder = file("${tempFolder}/npm")
+def yarnFolder = file("${tempFolder}/yarn")
+
 if (buildGutenbergMobileJSBundle) {
     println 'Building the Gutenberg Mobile JS bundle'
 
@@ -55,16 +60,14 @@ if (buildGutenbergMobileJSBundle) {
         // If false, it will try to use globally installed node.
         download = true
 
-        def tmpdir = "${System.getenv('TMPDIR')}/jsbundle/${System.getenv('VERSION')}"
-
         // Set the work directory for unpacking node
-        workDir = file("${tmpdir}/nodejs")
+        workDir = file(nodeFolder)
 
         // Set the work directory for NPM
-        npmWorkDir = file("${tmpdir}/npm")
+        npmWorkDir = file(npmFolder)
 
         // Set the work directory for Yarn
-        yarnWorkDir = file("${tmpdir}/yarn")
+        yarnWorkDir = file(yarnFolder)
 
         // Set the work directory where node_modules should be located
         nodeModulesDir = file("${project.projectDir}/../../")
@@ -290,10 +293,16 @@ If they are changed, the isBundleUpToDate flag is switched to false. That flag i
         delete '../../node_modules'
     }
 
+    task resetExtractedRNTools(type: Delete) {
+        println "Deleting temporary folders with extracted RN tools"
+        delete nodeFolder, npmFolder, yarnFolder
+    }
+
     preBuild.dependsOn(cleanupNodeModulesFolder)
     cleanupNodeModulesFolder.dependsOn(backupHermesDebugAAR)
     backupHermesDebugAAR.dependsOn(backupHermesReleaseAAR)
     backupHermesReleaseAAR.dependsOn(copyJSBundle)
     copyJSBundle.dependsOn(buildJSBundle)
     buildJSBundle.dependsOn(yarn_install)
+    nodeSetup.dependsOn(resetExtractedRNTools)
 }

